integration/tests/cook/test_basic.py

```diff
@@ -1106,6 +1116,10 @@ def test_error_while_creating_job(self):
         self.assertEqual(resp.status_code, 500)
 
     @pytest.mark.xfail
+    @unittest.skipIf(util.has_ephemeral_hosts(), 'If the cluster under test has ephemeral hosts, then it is generally '
+                                                 'a bad idea to use HOSTNAME EQUALS constraints, because it can cause '
+                                                 'the process responsible for launching hosts to launch hosts that '
+                                                 'never get used')
     def test_constraints(self):
```

Comment:
I know it's not part of your PR, but can you rename this so it's specifically referring to the hostname equals constraint? I was concerned we were turning off something more generic.

Reply:
Done